Sparkle/SPUDownloaderDeprecated.m

```diff
@@ -0,0 +1,184 @@
+//
+//  SPUDownloaderDeprecated.m
+//  Sparkle
+//
+//  Created by Deadpikle on 12/20/17.
+//  Copyright © 2017 Sparkle Project. All rights reserved.
+//
+
+#import "SPUDownloaderDeprecated.h"
+#import "SPUURLRequest.h"
+#import "SPUDownloader_Private.h"
+#import "SPUDownloadData.h"
+#import "SPULocalCacheDirectory.h"
+#import "SUErrors.h"
+
+#include "AppKitPrevention.h"
+
+@interface SPUDownloaderDeprecated () <NSURLDownloadDelegate>
+
+@property (nonatomic) NSURLDownload *download;
+@property (nonatomic) NSURLResponse *response;
+
+@end
+
+@implementation SPUDownloaderDeprecated
+
+@synthesize download = _download;
+@synthesize response = _response;
+
+- (void)startPersistentDownloadWithRequest:(SPUURLRequest *)request bundleIdentifier:(NSString *)bundleIdentifier desiredFilename:(NSString *)desiredFilename
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (self.download == nil && self.delegate != nil) {
+            // Prevent service from automatically terminating while downloading the update asynchronously without any reply blocks
+            [[NSProcessInfo processInfo] disableAutomaticTermination:SUDownloadingReason];
+            self.disabledAutomaticTermination = YES;
+            
+            self.mode = SPUDownloadModePersistent;
+            self.desiredFilename = desiredFilename;
+            self.bundleIdentifier = bundleIdentifier;
+            
+            self.download = [[NSURLDownload alloc] initWithRequest:request.request delegate:self];
+        }
+    });
+}
+
+- (void)startTemporaryDownloadWithRequest:(SPUURLRequest *)request
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (self.download == nil && self.delegate != nil) {
+            // Prevent service from automatically terminating while downloading the update asynchronously without any reply blocks
+            [[NSProcessInfo processInfo] disableAutomaticTermination:SUDownloadingReason];
+            self.disabledAutomaticTermination = YES;
+            
+            self.mode = SPUDownloadModeTemporary;
+            self.download = [[NSURLDownload alloc] initWithRequest:request.request delegate:self];
+        }
+    });
+}
+
+- (void)download:(NSURLDownload *)__unused d decideDestinationWithSuggestedFilename:(NSString *)name
+{
+    if (self.mode == SPUDownloadModeTemporary)
+    {
+        // Files downloaded in temporary mode should not last for very long,
+        // so it's ideal to place them in a system temporary directory
+        NSString *destinationFilename = NSTemporaryDirectory();
+        if (destinationFilename)
+        {
+            destinationFilename = [destinationFilename stringByAppendingPathComponent:name];
+            
+            [self.download setDestination:destinationFilename allowOverwrite:NO];
+        }
+    }
+    else
+    {
+        // Remove our old caches path so we don't start accumulating files in there
```

Comment:
The cache cleanup code is duplicated in both classes. Would you be able to move it to a method in the base class?